### ISSUE TITLE
how to use the input masks??

### ISSUE THREAD

**Issue by w1005444804** [facebookresearch/segment-anything#242]
As the title says,  when I followed the instructions, I didn't meet my expectations。
instructions： 
mask_input (np.ndarray): A low resolution mask input to the model, typically  coming from a previous prediction iteration. Has form 1xHxW, where  for SAM, H=W=256.

Can you provide an example？ thanks

**Comment by WusterHappy** [comment]
我也遇到同样的问题

**Comment by hungtooc** [comment]
yes, we need code to create custom mask

**Comment by hungtooc** [comment]
Update, I guess they preprocess mask by logits:
```python
def logit(x):
    a = torch.tensor(x)
    return torch.special.logit(a, eps=1e-6).numpy()
def inv_logit(x):
    return 0.5*(1. + np.sign(x)*(2./(1. + np.exp(-np.abs(x))) - 1.))

#load mask
mask = cv2.imread("images/4.png", cv2.IMREAD_GRAYSCALE)
mask = cv2.resize(mask, (256,256))
mask =  mask[None, :, :][None, :, :]
mask_logits = logit(mask)

```

**Comment by ThePassedWind** [comment]
> Update, I guess they preprocess mask by logits:
> 
> ```python
> def logit(x):
>     a = torch.tensor(x)
>     return torch.special.logit(a, eps=1e-6).numpy()
> def inv_logit(x):
>     return 0.5*(1. + np.sign(x)*(2./(1. + np.exp(-np.abs(x))) - 1.))
> 
> #load mask
> mask = cv2.imread("images/4.png", cv2.IMREAD_GRAYSCALE)
> mask = cv2.resize(mask, (256,256))
> mask =  mask[None, :, :][None, :, :]
> mask_logits = logit(mask)
> ```
> 
> this is my code, I'm not sure if it correct 100%

I tried but failed again. The output mask is nothing.

**Comment by antoniocandito** [comment]
hello, 
I am getting the same outcome, nothing if I input a binary mask insted points or bbox.

**Comment by ThePassedWind** [comment]
> hello, I am getting the same outcome, nothing if I input a binary mask insted points or bbox.

Hi,
I guess, it cannot support that the prompt is only mask, but with points.
Maybe the mask is just a secondary role

**Comment by cip8** [comment]
Any updates on this issue? Did you manage to make it work with a custom mask?

I tried converting my grayscale mask to SAM with this function:
```    def reference_mask_to_sam_mask(self, ref_mask: np.ndarray) -> np.ndarray:
        # Normalize the mask to [0, 1]
        ref_mask_norm = ref_mask / 255.0

        # Add a small constant to avoid extreme values
        epsilon = 1e-7
        ref_mask_norm = np.clip(ref_mask_norm, epsilon, 1 - epsilon)

        # Resize the mask to (256, 256)
        ref_mask_resized = resize(
            ref_mask_norm, (256, 256), mode="reflect", anti_aliasing=True
        )

        # Convert probabilities to logits
        ref_mask_logits = logit(ref_mask_resized)

        # Add an extra dimension to make the shape (1, 256, 256)
        sam_mask = ref_mask_logits[np.newaxis, :, :]

        return sam_mask
        ```
        
The output is not the one I expected, though - any ideas why?

Thanks a ton!! 💙🖖

**Comment by markushunter** [comment]
Similar thread: https://github.com/facebookresearch/segment-anything/issues/169

**Comment by markushunter** [comment]
This kinda works: https://github.com/facebookresearch/segment-anything/issues/169#issuecomment-1550222757

**Comment by Github-Yilei** [comment]
this might be useful: [Specifying a specific object with a box](https://github.com/facebookresearch/segment-anything/blob/main/notebooks/predictor_example.ipynb)

**Comment by lxr-1204** [comment]
I think that if it is the output MASK of a certain network, it may be taken out in the layer before the network output as low_mask
